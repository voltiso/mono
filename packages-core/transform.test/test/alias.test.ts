// â €â“¥ 2022     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import * as fs from 'node:fs/promises'

describe('alias', () => {
	it('works', async () => {
		expect.hasAssertions()

		const file = await fs.readFile('./dist/cjs/alias.d.ts')
		expect(file.toString()).toMatchSnapshot()
	})
})
