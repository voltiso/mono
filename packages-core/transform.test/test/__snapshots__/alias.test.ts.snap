// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alias works 1`] = `
"export declare type Flatten<T> = [{
    [k in keyof T]: T[k];
}][0];
/** @inline */ export declare type InlineFlatten<T> = [{
    [k in keyof T]: T[k];
}][0];
export declare type A = {
    a: 1;
};
export declare type B = {
    b: 1;
};
export declare type Normal = Flatten<A & B>;
export declare type Inlined = /*InlineFlatten<A & B>*/ { a: 1; b: 1; };
export declare type UnableToInline<B> = InlineFlatten<A & B>;
//# sourceMappingURL=alias.d.ts.map"
`;
